src/loaders/tiff/multi-tiff.ts

```diff
@@ -2,22 +2,55 @@ import type { GeoTIFFImage } from 'geotiff';
 
 import TiffPixelSource from './pixel-source';
 import { guessTiffTileSize } from '../utils';
-import { getMultiTiffMetadata, getMultiTiffMeta } from './lib/utils';
+import {
+  getMultiTiffMetadata,
+  getMultiTiffMeta,
+  OmeTiffSelection
+} from './lib/utils';
 import type Pool from './lib/Pool';
 import { getMultiTiffIndexer } from './lib/indexers';
-import type { TiffSelection } from './types';
 
 export interface MultiTiffImage {
   name: string;
-  selection: TiffSelection;
+  selection: OmeTiffSelection;
   tiff: GeoTIFFImage;
 }
 
+function assertSameResolution(images: MultiTiffImage[]) {
+  const width = images[0].tiff.getWidth();
+  const height = images[0].tiff.getHeight();
+  for (const image of images) {
+    if (image.tiff.getWidth() !== width || image.tiff.getHeight() !== height) {
+      throw new Error(`All images must have the same width and height`);
+    }
+  }
+}
+
+async function assertCompleteStack<T>(
```

Comment:
```suggestion
async function assertCompleteStack(
```

My fault for including this in the code snippet, but I don't think the generic parameter is used in the function definition.